Game/MyProject/Game/RoadGenerator/RoadEditor.cpp

```diff
@@ -0,0 +1,219 @@
+            point->setRenderable(meshInstanceRed);
+        else
+            point->setRenderable(meshInstanceRed);
+    }
+}
+
+
+void RoadEditor::getPointsFromScene()
+{
+    sceneGraph->getRoot()->findAllNodesContainingName("bezierPoint", bezierScenePoints);
+    
+    for(int i = 0; i < bezierScenePoints.size(); i++)
+        bezierPoints.push_back(bezierScenePoints[i]->getLocalPosition());
+}
+
+void RoadEditor::resizeRoadEditor()
+{
+    if(bezierPoints.size() != bezierScenePoints.size() || bezierMiddlePoints.size() == 0)
+    {
+        
+        for(int i = 0; i < bezierScenePoints.size(); i++)
+        {
+            bezierScenePoints[i]->remove();
+        }
+        bezierScenePoints.clear();
+        
+        
+        for(int i = 0; i < bezierMiddlePoints.size(); i++)
+        {
+            bezierMiddlePoints[i]->remove();
+        }
+        bezierMiddlePoints.clear();
+        
+        for(int i = 0; i < bezierMarginPoints.size(); i++)
+        {
+            bezierMarginPoints[i].first->remove();
+            bezierMarginPoints[i].second->remove();
+        }
+        bezierMarginPoints.clear();
+        
+        
+        for(int i = 0; i < rMiddlePoints.size(); i++)
+        {
+            std::shared_ptr<SceneNode> point = std::make_shared<SceneNode>("bezierMiddlePoint" + std::to_string(i));
+            sceneGraph->getRoot()->addChild(point);
+            point->setLocalPosition(rMiddlePoints[i]);
+            point->setRenderable(meshInstanceBlue);
+            bezierMiddlePoints.push_back(point);
+        }
+        
+        for(int i = 0; i < rMarginPoints.size(); i++)
+        {
+            std::shared_ptr<SceneNode> pointL = std::make_shared<SceneNode>("bezierMarginLPoint" + std::to_string(i));
+            sceneGraph->getRoot()->addChild(pointL);
+            pointL->setLocalPosition(rMarginPoints[i].first);
+            pointL->setRenderable(meshInstanceBlue);
+            
+            std::shared_ptr<SceneNode> pointR = std::make_shared<SceneNode>("bezierMarginRPoint" + std::to_string(i));
+            sceneGraph->getRoot()->addChild(pointR);
+            pointR->setLocalPosition(rMarginPoints[i].second);
+            pointR->setRenderable(meshInstanceBlue);
+            bezierMarginPoints.push_back(std::make_pair(pointL, pointR));
+        }
+        
+        //render bezier points at final to see the color
+        for(int i = 0; i < bezierPoints.size(); i++)
+        {
+            std::shared_ptr<SceneNode> point = std::make_shared<SceneNode>("bezierPoint" + std::to_string(i));
+            sceneGraph->getRoot()->addChild(point);
+            point->setLocalPosition(bezierPoints[i]);
+            
+            if(i % 3 == 0)
+                point->setRenderable(meshInstanceRed);
+            else
+                point->setRenderable(meshInstanceGreen);
+            
+            bezierScenePoints.push_back(point);
+        }
+        
+    }
+}
+
+void RoadEditor::buildRoadEditor()
```

Comment:
This functions creates some nodes as far if i understand right and updates the position. You should split this function in two build/update positions. Also avoid calling this on each frame. call updates only when needed.